redistore.go

```diff
@@ -137,23 +137,30 @@ func (s *RediStore) SetMaxAge(v int) {
 	}
 }
 
-func dial(network, address, password string) (redis.Conn, error) {
+func dial(network, address, username string, password string) (redis.Conn, error) {
 	c, err := redis.Dial(network, address)
 	if err != nil {
 		return nil, err
 	}
 	if password != "" {
-		if _, err := c.Do("AUTH", password); err != nil {
-			c.Close()
-			return nil, err
+		if username != "" {
+			if _, err := c.Do("AUTH", password); err != nil {
+				c.Close()
+				return nil, err
+			}
+		} else {
+			if _, err := c.Do("AUTH", username, password); err != nil {
+				c.Close()
+				return nil, err
+			}
```

Comment:
You switched the call in the condition:
 - empty `username` calls the command using the `username` variable
 - a non-empty `username` calls the command without using the `username` variable

Reply:
Yes, sorry for the mistaken.

I don't think this PR will be accepted by the author.
The repo is inactive for such a long time.

Reply:
Besides, I suggest that using tcp link url to include username and database index is the trick to pass through this dilemma if your library is dependent on `redistore`.

Reply:
> Yes, sorry for the mistaken.
> 
> I don't think this PR will be accepted by the author.
> The repo is inactive for such a long time.

Yes, I am of the same opinion. Perhaps an authentik should use a different library than this one if we want to implement modern Redis features.